lib/Component/VCalendar.php

```diff
@@ -486,5 +494,26 @@ function validate($options = 0) {
 
     }
 
-}
+    /**
+     * Returns all components with a specific UID value.
+     *
+     * @return array
+     */
+    function getByUID($uid) {
+
+        return array_filter($this->children, function($item) use ($uid) {
 
+            if (!$item instanceof Component) {
```

Comment:
can all known components have a `UID` or can we filter more agressive based on the class here?